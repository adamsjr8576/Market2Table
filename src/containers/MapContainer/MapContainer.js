import React, { Component } from 'react';
import { Map, GoogleApiWrapper } from 'google-maps-react';
import { InfoWindow, Marker } from 'google-maps-react';

const mapStyles = {
  width: '700px',
  height: '500px',
  borderRadius: '20px',
  border: '3px solid black'
};

export class MapContainer extends Component {
  constructor() {
    super()
    this.state = {
      showingInfoWindow: false,
      activeMarker: {},
      selectedPlace: {}
    }
  }

  onMarkerClick = (props, marker, e) =>
   this.setState({
     selectedPlace: props,
     activeMarker: marker,
     showingInfoWindow: true
   });

  onClose = props => {
   if (this.state.showingInfoWindow) {
     this.setState({
       showingInfoWindow: false,
       activeMarker: null
     });
   }
  };

  setMarkers = () => {
    if (this.props.markets === undefined) {
      const { market } = this.props
      return (<Marker
          onClick={this.onMarkerClick}
          name={market.marketname}
          position={{ lat: market.latitude, lng: market.longitude }}
          key={market.id}
        />)
    } else {
      const { markets } = this.props
      return markets.map(market => {
        return (
          <Marker
              onClick={this.onMarkerClick}
              name={market.marketname}
              position={{ lat: market.latitude, lng: market.longitude }}
              key={market.id}
            />
        )
      })
    }
  }

  setCenter = () => {
    if (this.props.markets === undefined) {
      const { market } = this.props
      return {
        lat: this.props.market.latitude,
        lng: this.props.market.longitude
      }
    } else {
      const { markets } = this.props
      return {
       lat: markets[0].latitude,
       lng: markets[0].longitude
      }
    }
  }

  render() {
    const markers = this.setMarkers();
    const center = this.setCenter();
    console.log(center);
    return (
      <Map
        google={this.props.google}
        zoom={11}
        style={mapStyles}
        initialCenter={center}
      >
        {markers}
        <InfoWindow
          marker={this.state.activeMarker}
          visible={this.state.showingInfoWindow}
          onClose={this.onClose}
        >
          <div>
            <h4>{this.state.selectedPlace.name}</h4>
          </div>
        </InfoWindow>
      </Map>
    );
  }
}

export default GoogleApiWrapper({
  apiKey: 'AIzaSyBOCAFIBU1dbEoO0GYU73BhetuwHLzM5gA'
})(MapContainer);
