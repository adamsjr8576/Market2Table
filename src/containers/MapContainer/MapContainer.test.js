import React from 'react';
import {shallow} from 'enzyme';
import { MapContainer } from './MapContainer';

describe('MapContainer Component', () => {
  it('should match the snapshot with market and no path passed in', () => {
    const mockMarket = {
      id: "1002192",
      marketname: "Ballston FRESHFARM Market",
      favorite: false,
      GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
      Address: "901 N Taylor St, Ballston, Virginia, 22203",
      Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
      Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
      latitude: "38.881112",
      longitude: "-77.112179"
    }
    const wrapper = shallow(<MapContainer market={mockMarket} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with markets and markets path passed in', () => {
    const mockMarkets = [{
      id: "1002192",
      marketname: "Ballston FRESHFARM Market",
      favorite: false,
      GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
      Address: "901 N Taylor St, Ballston, Virginia, 22203",
      Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
      Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
      latitude: "38.881112",
      longitude: "-77.112179"
    },
    {
      id: "22008839",
      marketname: "Ballston Farmers Market",
      favorite: false,
      GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
      Address: "901 N Taylor St, Ballston, Virginia, 22203",
      Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
      Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
      latitude: "38.881112",
      longitude: "-77.112179"
    }]
    const wrapper = shallow(<MapContainer markets={mockMarkets} path="/markets" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with markets and favorites path passed in', () => {
    const mockMarkets = [{
      id: "1002192",
      marketname: "Ballston FRESHFARM Market",
      favorite: false,
      GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
      Address: "901 N Taylor St, Ballston, Virginia, 22203",
      Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
      Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
      latitude: "38.881112",
      longitude: "-77.112179"
    },
    {
      id: "22008839",
      marketname: "Ballston Farmers Market",
      favorite: false,
      GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
      Address: "901 N Taylor St, Ballston, Virginia, 22203",
      Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
      Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
      latitude: "38.881112",
      longitude: "-77.112179"
    }]
    const wrapper = shallow(<MapContainer markets={mockMarkets} path="/favorites" />);
    expect(wrapper).toMatchSnapshot();
  });

});
