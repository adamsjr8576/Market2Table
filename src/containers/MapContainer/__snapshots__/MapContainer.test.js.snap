// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer Component should match the snapshot with market and no path passed in 1`] = `
<div
  id="mapBox"
>
  <Map
    center={Object {}}
    centerAroundCurrentLocation={false}
    containerStyle={Object {}}
    initialCenter={
      Object {
        "lat": "38.881112",
        "lng": "-77.112179",
      }
    }
    style={
      Object {
        "border": "3px solid black",
        "borderRadius": "20px",
        "height": "480px",
        "width": "680px",
      }
    }
    visible={true}
    zoom={11}
  >
    <Marker
      key="1002192"
      name="Ballston FRESHFARM Market"
      onClick={[Function]}
      position={
        Object {
          "lat": "38.881112",
          "lng": "-77.112179",
        }
      }
    />
    <InfoWindow
      marker={Object {}}
      onClose={[Function]}
      visible={false}
    >
      <div>
        <h4 />
      </div>
    </InfoWindow>
  </Map>
</div>
`;

exports[`MapContainer Component should match the snapshot with markets and favorites path passed in 1`] = `
<div
  id="mapBox"
>
  <Map
    center={Object {}}
    centerAroundCurrentLocation={false}
    containerStyle={Object {}}
    initialCenter={
      Object {
        "lat": "39.270018",
        "lng": "-97.970908",
      }
    }
    style={
      Object {
        "border": "3px solid black",
        "borderRadius": "20px",
        "height": "480px",
        "width": "680px",
      }
    }
    visible={true}
    zoom={4}
  >
    <Marker
      key="1002192"
      name="Ballston FRESHFARM Market"
      onClick={[Function]}
      position={
        Object {
          "lat": "38.881112",
          "lng": "-77.112179",
        }
      }
    />
    <Marker
      key="22008839"
      name="Ballston Farmers Market"
      onClick={[Function]}
      position={
        Object {
          "lat": "38.881112",
          "lng": "-77.112179",
        }
      }
    />
    <InfoWindow
      marker={Object {}}
      onClose={[Function]}
      visible={false}
    >
      <div>
        <h4 />
      </div>
    </InfoWindow>
  </Map>
</div>
`;

exports[`MapContainer Component should match the snapshot with markets and markets path passed in 1`] = `
<div
  id="mapBox"
>
  <Map
    center={Object {}}
    centerAroundCurrentLocation={false}
    containerStyle={Object {}}
    initialCenter={
      Object {
        "lat": "38.881112",
        "lng": "-77.112179",
      }
    }
    style={
      Object {
        "border": "3px solid black",
        "borderRadius": "20px",
        "height": "480px",
        "width": "680px",
      }
    }
    visible={true}
    zoom={11}
  >
    <Marker
      key="1002192"
      name="Ballston FRESHFARM Market"
      onClick={[Function]}
      position={
        Object {
          "lat": "38.881112",
          "lng": "-77.112179",
        }
      }
    />
    <Marker
      key="22008839"
      name="Ballston Farmers Market"
      onClick={[Function]}
      position={
        Object {
          "lat": "38.881112",
          "lng": "-77.112179",
        }
      }
    />
    <InfoWindow
      marker={Object {}}
      onClose={[Function]}
      visible={false}
    >
      <div>
        <h4 />
      </div>
    </InfoWindow>
  </Map>
</div>
`;
